### PR TITLE
EVM pallet

### PR DESCRIPTION
**Pull request by Eve-Parity**
Took EVM content from the legacy `runtime/smart-contracts.md` file and added it to its own topic.

### CODE REVIEW

File: source/docs/main-docs/05-design/evm-pallet.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+Substrate is built to enable you to extend what's provided out of the box. We encourage further development of alternative smart contract platforms on top of the Substrate runtime. Use these pre-built pallets to inform how you might design your own system or how you could port over an existing system to work on a Substrate-based chain.
+
+## FAQ
+
+### What is the difference between memory and storage?
+
+In ink! `memory` is the computer memory that is commonly known to programmers, while `storage` is the contract instance memory. The `storage` is backed up by the runtime in a database. <!--Accesses to it are considered to be slow.-->
+
+### How do I run tests?
```
Comment (NukeManDan) [inline source/docs/main-docs/05-design/evm-pallet.md:48]: why is ink in here? this is explicitly different than frontier and EVM, should live in its own page and cross reference each other

**Review by NukeManDan — CHANGES_REQUESTED** [review]
ink and contracts pallet should be separate from frontier and evm.

File: source/docs/main-docs/05-design/evm-pallet.md (hunk 1 above)
Comment (sacha-l) [reply source/docs/main-docs/05-design/evm-pallet.md:48]: Yes -- to add to Dan's point: the EVM pallet has nothing to do with writing smart contracts (and thereby, tests). All it's used for is to execute smart contracts that compile to EVM byte code.

**Review by sacha-l — APPROVED** [review]
Youpeee - one step closer to the finish line 🏁

**Comment by Eve-Parity** [comment]
I'm a bit concerned with some mixed messages in this content. At one point we note that EVM is not a good idea in production using modern equipment, yet we're explaining how to use it. Can I get some clarification on this @NukeManDan @sacha-l ?

**Comment by shawntabrizi** [comment]
> At one point we note that EVM is not a good idea in production using modern equipment

I think you may be misinterpreting the comment about 256 bit integers. It was a decision made about the EVM which absolutely an overhead decision when running on mainly 32/64 bit architectures, however doesn't mean users shouldn't use it.

EVM is obviously a very useful environment, and there is not much we can do at this point to change how its fundamental architecture is designed.

File: source/docs/main-docs/05-design/evm-pallet.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
+# EVM pallet
+
+The FRAME Ethereum Virtual Machine (EVM) provides an execution environment for Substrate's Ethereum compatibility layer, known as Frontier. Frontier allows unmodified EVM code to be executed in a Substrate-based blockchain, designed to closely emulate the functionality of executing contracts on the Ethereum mainnet within the Substrate runtime. 
+
+For more information on the FRAME EVM, see [FRAME EVM pallet reference](https://docs.rs/pallet_evm/).
+
+To download Frontier, go to https://github.com/paritytech/frontier.
```
Comment (sacha-l) [inline source/docs/main-docs/05-design/evm-pallet.md:7]:
```suggestion
To use the Frontier node template, go to https://github.com/paritytech/frontier/tree/master/template.
```

**Review by sacha-l — APPROVED** [review]

**Comment by sacha-l** [comment]
Merging this as is and will follow-up with a new PR to refactor and reorganize the content here.

**Pull request merged**